minio/credentials.py

```diff
@@ -0,0 +1,256 @@
+# -*- coding: utf-8 -*-
+# Minio Python Library for Amazon S3 Compatible Cloud Storage, (C) 2016, 2017 Minio, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+minio.credentials
+~~~~~~~~~~~~~~~
+
+This module contains :class:` credentials` implementation.
+
+:copyright: (c) 2016, 2017 by Minio, Inc.
+:license: Apache 2.0, see LICENSE for more details.
+
+"""
+
+from abc import ABCMeta, abstractmethod
+
+import os, json, urllib3, datetime
+import ConfigParser
+
+from .error import MinioError
+
+
+# credentials class
+class credentials(object):
+
+    def __init__(self, access_key, secret_key, session_token):
+        self.access_key = access_key
+        self.secret_key = secret_key
+        self.session_token = session_token
+
+# credentials_provider class
+class credentials_provider(object):
+
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def is_expired(): pass
+
+    @abstractmethod
+    def retrieve(): pass 
+
+    def __init__(self):
+        self._force_refresh = False
+        self._cached_creds = None
+
+    def get(self):
+        if self._force_refresh or self.is_expired():
+            self._cached_creds = self.retrieve()
+            self._force_refresh = False
+        return self._cached_creds
+
+    def expire(self):
+        self._force_refresh = True
+
+# static_credentials
+class static_credentials(credentials_provider):
+
+    def __init__(self, access_key, secret_key, session_token=""):
+        super(static_credentials, self).__init__()
+        self._access_key = access_key
+        self._secret_key = secret_key
+        if session_token == None:
```

Comment:
`session_token` will never be `None`, unless you strictly specify it

 if it is `None` then you shouldn't supply it to a method call

Just have...
```python
_session_token = session_token
```